torch/lib/TH/generic/THTensor.c

```diff
@@ -285,23 +285,103 @@ void THTensor_(resize5d)(THTensor *self, long size0, long size1, long size2, lon
   THTensor_(resizeNd)(self, 5, size, NULL);
 }
 
-THTensor* THTensor_(newExpand)(THTensor *tensor, THLongStorage *sizes) {
-  THArgCheck(THLongStorage_size(sizes) >= THTensor_(nDimension)(tensor), 1, "the number of sizes provided \
-      must be greater or equal to the number of dimensions in the tensor");
-  THArgCheck(THTensor_(nDimension)(tensor) > 0, 0, "can't expand an empty tensor");
+THTensor* THTensor_(newExpand)(THTensor *tensor, THLongStorage *sizes, int raiseErrors) {
```

Comment:
Yes, it is guaranteed in PyTorch. We depend on it all over the place (see how `HANDLE_TH_ERRORS` macro is defined in `torch/csrc/Exceptions.h`). I don't think having this flag as part of the interface is a good idea. It's only for this single use case.